src/Downloader.ts

```diff
@@ -126,12 +129,27 @@ export class Downloader {
       await this.onProgress(downloaded, fileMeta.size);
     }
     const decryptedHash = hasher2Hex(this.hasher.digest());
-    const success = fileMeta.hash == decryptedHash;
+    const success = fileMeta.hash === decryptedHash;
     if (success) {
-      fileWriter.createDownload();
+      const out = await fileWriter.createDownload();
       await this.onSuccess();
+      return out
     } else {
       throw new Error('Hash does not matches with uploaded file');
     }
   };
+
+  // TODO - Remove during API breaking changes
+  // @deprecated - Use downloadToFilesystem
+  public download (did) {
+    return this._download(did, 'download')
+  }
+
+  public downloadToFilesystem (did): Promise<void> {
+    return this._download('download', 'download') as Promise<void>
```

Comment:
shouldn't this be
```js
    return this._download(did, 'download') as Promise<void>
```

Reply:
Looks like a horrible mistake, fixing ASAP.

Reply:
Fixed.